src/wpantund/NCPInstanceBase-Addresses.cpp

```diff
@@ -775,8 +775,7 @@ NCPInstanceBase::multicast_address_was_joined(Origin origin, const struct in6_ad
 		if ((origin == kOriginThreadNCP) || (origin == kOriginUser)) {
 			mPrimaryInterface->join_multicast_address(&address);
 		}
-
-		if (origin == kOriginPrimaryInterface) {
+		if (origin == kOriginPrimaryInterface && !ncp_state_is_initializing(get_ncp_state())) {
```

Comment:
Not sure about not allowing addresses to be updated while NCP is being initalized.
With this we can potentially miss adding a unicast/multicast address added on interface (causing a miss-match between interface and NCP).